docs/content/guides/dagster/branch_depl_with_snowflake.mdx

```diff
@@ -0,0 +1,325 @@
+
+```python file=/guides/dagster/local_to_prod_tutorial/branch_depl_tutorial/repository_v2.py startafter=start_resources endbefore=end_resources
+resource_defs = {
+    "branch": {
+        "hn_client": hn_api_client,
+        "snowflake_io_mgr": snowflake_io_manager.configured(
+            {
+                **snowflake_config,
+                "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+            }
+        ),
+        "snowflake": snowflake_resource.configured(
+            {
+                **snowflake_config,
+                "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+            }
+        ),
+    },
+    "production": {
+        "hn_client": hn_api_client,
+        "snowflake_io_mgr": snowflake_io_manager.configured(
+            {
+                **snowflake_config,
+                "database": "PRODUCTION",
+            }
+        ),
+        "snowflake": snowflake_resource.configured({**snowflake_config, "database": "PRODUCTION"}),
+    },
+}
+```
+
+Then, we can add the `clone_prod` job to our repository:
+
+```python file=/guides/dagster/local_to_prod_tutorial/branch_depl_tutorial/repository_v2.py startafter=start_repository endbefore=end_repository
+@repository
+def repo():
+    ...
+    return [
+        with_resources([items, comments, stories], resource_defs=resource_defs[get_current_env()]),
+        clone_prod.to_job(resource_defs=resource_defs[get_current_env()]),
+    ]
+```
+
+In our `branch_deployments.yml` file, we’ll add an additional step to the existing `dagster_cloud_build_push` job that queues a run of `clone_prod` within each branch deployment upon launch. The `if` condition below ensures that `PRODUCTION` will not be cloned if the pull request is closed:
+
+```yaml file=/guides/dagster/local_to_prod_tutorial/branch_depl_tutorial/clone_prod.yaml
+name: Dagster Branch Deployments
+  on:
+    pull_request:
+      types: [opened, synchronize, reopened, closed]
+  env:
+    DAGSTER_CLOUD_URL: ${{ secrets.DAGSTER_CLOUD_URL }}
+    REGISTRY_URL: ${{ secrets.REGISTRY_URL }}
+
+  jobs:
+    dagster_cloud_build_push:
+      runs-on: ubuntu-latest
+      name: Dagster Branch Deployments
+      strategy:
+        ...
+      steps:
+        ...
+        - name: Clone Snowflake schema upon launch
+          if: github.event.action != 'closed'
+          uses: dagster-io/cloud-branch-deployments-action/run@main
+          with:
+            location: ${{ toJson(matrix.location) }}
+            deployment: ${{ steps.deploy.outputs.deployment }}
+            repository: repo
+            job: clone_prod
+          env:
+            DAGSTER_CLOUD_API_TOKEN: ${{ secrets.DAGSTER_CLOUD_API_TOKEN }}
+```
+
+Opening a pull request for our current branch will automatically kick off a branch deployment. After the deployment launches, we can confirm that the `clone_prod` job is run:
```

Comment:
does the db reclone when you push updates to the PR?

Reply:
+1

Reply:
updated the doc to make this more clear!